### ISSUE TITLE
suggest: add scripting define so that things can be used in non-strange projects as well

### ISSUE THREAD

**Issue by iwaldrop** [strangeioc/strangeioc#207]
## use case
In strange you'd inject a dependency into an object, but if you were to take that class into an non-strange project then you'd get a compile error on the usage of the `Inject` attribute.

## summary
Introduce a scripting define symbol into the project such as, `USE_STRANGE_IOC` which would be added automatically via an editor script when the package is installed, and optionally enabled/disabled via menu items / preferences pane.

**Comment by leandromperez** [comment]
Here is a thought..Wouldn't it be simpler if you just define that attribute
class in the non-strange project?

**Comment by unitymarc** [comment]
Hmmm...well, I take both points. This might be a better idea for the "plug-ins" repo we keep talking about making. I.e., have a standard way for Strange classes to include an opt-out. It's not really "part of Strange", but could be useful whenever someone wanted to make a class work on either side of that particular divide.

**Comment by iwaldrop** [comment]
Yeah, just having a dummy `InjectAttribute` class to throw in projects was my first thought, but then there's the issue of distributing stuff. You either include your dummy in the distro which would conflict with a target where strange was being used, or you don't and point to it in a readme. Either way there's some gotchas with onboarding.
